util/encoding/encoding.go

```diff
@@ -50,6 +45,13 @@ const (
 
 	bytesMarker byte = floatInfinity + 1
 	timeMarker  byte = bytesMarker + 1
+
+	// intMin is chosen such that the range of int tags does not overlap the
+	// ascii character set that is frequently used in testing.
+	intMin   = 0x80
+	intZero  = intMin + 8
+	intSmall = 111
```

Comment:
Where did this come from? Is it the value that makes `intMax` equal to `0xff`? If so maybe they should go the other way:
```
intMax = 0xff
intSmall = intMax - intZero - 8
```